### PR TITLE
Refactored `DropSourceBehavior` to partial and split `BehaviorDataObject` into its own file

### PR DESCRIPTION
**Pull request by elachlan**
Refactored `DropSourceBehavior` to partial file and split `BehaviorDataObject` into its own file.

Enabled nullability in `BehaviorDataObject`.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8531)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.BehaviorDataObject.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+using System.Collections;
+using System.ComponentModel;
+
+namespace System.Windows.Forms.Design.Behavior
+{
+    internal sealed partial class DropSourceBehavior
+    {
+        /// <summary>
+        ///  This class extends from DataObject and carries additional
```
Comment (dmitrii-drobotov) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.BehaviorDataObject.cs:13]:
```suggestion
        ///  This class extends from <see cref="DataObject"/> and carries additional
```

**Review by dmitrii-drobotov — APPROVED** [review]
Looks good, I only have a minor suggestion.

**Review by lonitra — DISMISSED** [review]
LGTM, once Dmitrii's comment has been addressed we can get this merged.

**Comment by elachlan** [comment]
@lonitra done.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.BehaviorDataObject.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                _dragComponents = dragComponents;
+                _source = source;
+                _sourceBehavior = sourceBehavior;
+                _target = null;
+            }
+
+            public Control Source
+            {
+                get => _source;
+            }
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.BehaviorDataObject.cs:34]:
```suggestion
            public Control Source => _source;
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.BehaviorDataObject.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+            public ICollection DragComponents
+            {
+                get => _dragComponents;
+            }
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.BehaviorDataObject.cs:39]:
```suggestion
            public ICollection DragComponents => _dragComponents;
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.BehaviorDataObject.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+            public IComponent? Target
+            {
+                get => _target;
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DropSourceBehavior.BehaviorDataObject.cs:43]: do you need this backing field `_target` ?

**Review by dreddy-work — COMMENTED** [review]
as you moved class to new file, i added some nits to clean it up.

**Review by dmitrii-drobotov — APPROVED** [review]

**Review by lonitra — APPROVED** [review]

**Pull request merged**